hs_discover/views.py

```diff
@@ -1,57 +1,224 @@
+            })
+
+        asc = '-1'
+        if request.GET.get('asc'):
+            asc = request.GET.get('asc')
+
+        sort = 'modified'
+        if request.GET.get('sort'):
+            sort = request.GET.get('sort')
+        sort = sort if asc == '1' else '-{}'.format(sort)
+
+        if request.GET.get('q'):
+            q = request.GET.get('q')
+            sqs = sqs.filter(content=q)  # .boost('keyword', 2.0)
 
-        sqs = SearchQuerySet()
+        try:
+            qs = request.query_params
+            filters = json.loads(qs.get('filter'))
+            # filter values expect lists, for example discoverapi/?filter={"owner":["Firstname Lastname"]}
+            if filters.get('author'):
+                sqs = sqs.filter(author__in=filters['author'])
+            if filters.get('owner'):
+                sqs = sqs.filter(owner__in=filters['owner'])
+            if filters.get('subject'):
+                sqs = sqs.filter(subject__in=filters['subject'])
+            if filters.get('contributor'):
+                sqs = sqs.filter(contributor__in=filters['contributor'])
+            if filters.get('type'):
+                sqs = sqs.filter(resource_type__in=list(filters['type']))
+            if filters.get('availability'):
+                sqs = sqs.filter(availability__in=filters['availability'])
+            if filters.get('geofilter'):
+                sqs = sqs.filter(north__range=[-90, 90])
+            if filters.get('date'):
+                try:
+                    datefilter = DateRange(start=datetime.datetime.strptime(filters['date'][0], '%Y-%m-%d'),
+                                           end=datetime.datetime.strptime(filters['date'][1], '%Y-%m-%d'))
 
-        vocab = []
-        for result in sqs:
-            if result.title:
-                vocab.extend(result.title.split(' '))
-            if result.subject:
-                vocab.extend(result.subject)
+                    # (datefilter.start < start_date < datefilter.end) or (start_date < datefilter.start)
+                    sqs = sqs.exclude(start_date__gt=datefilter.end).exclude(end_date__lt=datefilter.start)
+                except ValueError as e:
+                    pass  # ignore bad values and don't filter which is the correct action
+        except TypeError as type_ex:
+            pass  # no filters passed "the JSON object must be str, bytes or bytearray not NoneType"
 
-        vocab = [x for x in vocab if len(x) > 2]
-        vocab = list(set(vocab))
-        vocab = sorted(vocab)
+        except json.JSONDecodeError as parse_ex:
+            return JsonResponse({'message': 'Filter JSON parsing error - {}'.format(str(parse_ex)),
+                                 'received': request.query_params}, status=400)
 
-        if q:
-            sqs = sqs.filter(content=q).boost('keyword', 2.0)
+        except Exception as gen_ex:
+            logger.debug('hs_discover API - {}: {}'.format(type(gen_ex), str(gen_ex)))
```

Comment:
Warn is more appropriate than debug here.  This is an area we don't expect to hit, so when it does get hit we would like to know and have a record of it in the logs.

Reply:
Commit d41715b42